### ISSUE TITLE
Question of CityGaussianV2 DGD implementation

### ISSUE THREAD

**Issue by Sunnyhong0326** [DekuLiuTesla/CityGaussian#75]
Hi, thanks for your nice work!
I'm currently reading your CityGaussianV2 work but have a question about the DGD implementation.
From Figure2 of your paper, the densifcation threshold is set as positional gradient w.r.t D-SSIM Loss.
Does this means that in the implementation part, we need to backward pass the Loss 2 times to calculate gradient?
First backward pass is for calculating total loss and then gradient descent, second backward pass is for calculating D-SSSIM loss only for calculate the positional gradient w.r.t. D-SSIM loss.
![image](https://github.com/user-attachments/assets/0a61c20f-c2d0-4953-937c-9b4dec03383f)

**Comment by DekuLiuTesla** [comment]
Hi, @Sunnyhong0326, it's a good issue. In the implementation, the loss is separated into two parts and indeed backwarded two times. Here is the code snippet for reference:
![image](https://github.com/user-attachments/assets/a3b39072-4dd2-4ac2-8909-42d333db153f)

**Comment by Sunnyhong0326** [comment]
Thanks! 
I'm also curious that if backwarded two times, does it effect training time significantly?

**Comment by DekuLiuTesla** [comment]
Yes, it brings about an extra 10 minutes, but it is compensated after we remove redundant points in parallel tuning.
